maxim/logger/writer.py

```
@@ -366,24 +405,79 @@ def flush_log_files(self):
+                        logs = f.read()
+                except Exception as e:
+                    scribe().warning(
+                        f"[MaximSDK] Failed to read spilled log file {file}. Error: {e}"
+                    )
+                    continue
                 try:
                     self.maxim_api.push_logs(self.config.repository_id, logs)
-                    os.remove(os.path.join(self.logs_dir, file))
+                    os.remove(filepath)
+                    self.__file_failures.pop(file, None)
                 except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)